src/sentry/tasks/commit_context.py

```diff
@@ -46,8 +49,15 @@ def queue_comment_task_if_needed(commit: Commit, group_owner: GroupOwner):
         not pr.pullrequestcomment_set.exists()
         or group_owner.group_id not in pr.pullrequestcomment_set.get().group_ids
     ):
-        # TODO: Debouncing Logic
-        comment_workflow.delay(pullrequest_id=pr.id, project_id=group_owner.project_id)
+        lock = locks.get(f"queue_comment_task:{pr.id}", duration=10, name="queue_comment_task")
+        with lock.acquire():
+            cache_key = DEBOUNCE_PR_COMMENT_CACHE_KEY(pullrequest_id=pr.id)
+            if cache.get(cache_key) is not None:
```

Comment:
What kind of cache are we talking about? Is it possible that the task was in queue but it's expired and that's why we get None here? Or is your code below the only way something can get deleted?

Reply:
we have a cache util https://github.com/getsentry/sentry/blob/master/src/sentry/utils/cache.py. the cache returns `None` if the key doesn't exist, which should happen if it gets deleted in the task or if the key expires